ocfweb/docs/views/servers.py

```diff
@@ -57,81 +56,90 @@ def has_munin(self):
-                for hostname in (
-                    'fallingrocks',
-                    'monsoon',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='jaws',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'anthrax',
-                    'biohazard',
-                    'death',
-                    'dementors',
-                    'democracy',
-                    'firestorm',
-                    'flood',
-                    'lightning',
-                    'maelstrom',
-                    'pestilence',
-                    'reaper',
-                    'supernova',
-                    'thunder',
-                    'tsunami',
-                    'werewolves',
-                    'whirlwind',
-                    'whiteout',
-                    'zombies',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='pandemic',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'pileup',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='riptide',
-            type='hypervisor',
-        ),
-
-        Host.from_ldap(
-            hostname='corruption',
-            type='server',
-        ),
-
-        Host('blackhole', 'network', 'Managed Cisco Catalyst 2960S-48TS-L Switch.', []),
-
-        Host('logjam', 'printer', '', []),
-        Host('pagefault', 'printer', '', []),
-        Host('papercut', 'printer', '', []),
-        Host.from_ldap('overheat', type='raspi'),
-        Host.from_ldap('tornado', type='nuc'),
-
-        *(Host.from_ldap(desktop, type='desktop')
-          for desktop in sorted(list_desktops())),
-    ]
+    def create_hosts(lst):
+        """Accepts a list of raw ldap output, returns a dictionary of host
+        objects indexed by hostname
+        """
+        hosts = {}
+        for h in lst:
+            if not is_hidden(h):
+                description = h.get('description', [''])[0]
+                hostname = h['cn'][0]
+                hosts[hostname] = (Host(hostname, h['type'], description, ()))
+        return hosts
+
+    servers = create_hosts(hosts_by_filter('(type=server)'))
+    desktops = create_hosts(hosts_by_filter('(type=desktop)'))
+    misc = create_hosts(hosts_by_filter('(type=printer)'))
+    hypervisors = {}
+
+    physical_hosts = query_puppet(TEMP_IS_VIRT_FALSE)
+    all_children = query_puppet(TEMP_GET_CHILDREN)
+
+    # Add children to hypervisors
+    for h in list(servers.values()):
+        if h.hostname in physical_hosts:
+            # Populate a list of children
+            children = []
+            for child_hostname in all_children.get(h.hostname, []):
+                child = servers.get(child_hostname, False)
```

Comment:
It's slightly more idiomatic to just say `child = servers.get(child_hostname)` here. If there is no entry with key `child_hostname`, then `child` will be set to `None`, which is false in boolean context.